docs/deploy-full-rp-service-in-dev.md

```diff
@@ -3,31 +3,56 @@
 ## Prerequisites
 
 1. Your development environment is prepared according to the steps outlined in [Prepare Your Dev Environment](./prepare-your-dev-environment.md)
-
+2. During the deployment, don't edit files in your ARO-RP repository so that
+   `git status` reports a clean working tree. Otherwise aro container image
+   will have `-dirty` suffix, which can cause problems esp. when the repository
+   becomes dirty during the procedure.
```

Comment:
I guess we need to prevent this programmatically in this scenario because it can't be wanted. Unless somebody wanna deploy an RP using some not committed changes.
I would add some checks or warning messages to the scripts we use or _make_'s targets.
But we can do it in another PR. Does it make sense?

Reply:
What sort of problems does the -dirty suffix cause? Is it only an issue when deploying a shared rp or is it an issue for local dev as well?

Reply:
One solution would be to set COMMIT/TAG = latest if RP_MODE == development, this will make reruns or testing a lot easier.

Reply:
To me it's more of a recommendation. You can definitely deploy with the dirty tag if you want (I do it and sometimes it's an accident but it still works), but you will have the potential issues mentioned above.

Should we just change the wording to "it's recommended to..." and add the content here to the document? https://github.com/Azure/ARO-RP/pull/2964/files#r1238448125

Reply:
I updated the text as suggested by @cadenmarchese .